### PR TITLE
Add smoke test for altlayer config

### LINKED ISSUE #56: Add E2E tests for partner architectures
We want to make sure that our nitro integration can be deployed by our partners. To do that we will deploy test-node mimicking our partners configure their nitro architecture.

- [x] Ask for details about the nitro architecture choices from our partners.
- [ ] Ensure we have a way to run a docker compose environment with those configurations.
- [ ] Test these configurations in CI (like the current smoke tests).
- [ ] Add some tooling to deploy these configurations on top of our decaf testnet and run that at least once.
- [ ] Document how to run the deployment again so we can do it on demand.

We explicitly are not looking to do a persistent deployment so it's enough to start the nodes, assert they are working and shut them down. We might also run them for longer periods (days).

### PR DESCRIPTION
**Pull request by Sneh1999**
Closes #56

https://github.com/EspressoSystems/nitro-testnode/issues/56
### This PR:

Adds a smoke test with altlayer configuration

### CODE REVIEW

File: altlayer-test-node.bash [hunk 1]
```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/env bash
```
Comment (sveitser) [inline altlayer-test-node.bash]: How hard would it be to not duplicate this bash script and the docker compose file?

Comment (Sneh1999) [reply altlayer-test-node.bash]: Shouldnt be hard, I can do that!

Comment (Sneh1999) [reply altlayer-test-node.bash]: https://github.com/EspressoSystems/nitro-testnode/pull/59/commits/b41364dfef60af181ab8c13a5bfe43e4e2ec7390

**Review by ImJeremyHe — APPROVED** [review]

File: smoke-test-altlayer.bash [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+
+#  Run altlayer config with batch poster, sequencer, full node and validator
+./test-node.bash --init-force --validate --batchposters 1 --latest-espresso-image --detach 
+
+docker compose up -d full-node --detach
+
+# Sending L2 transaction through the full-node's api
+user=user_l2user
+./test-node.bash script send-l2 --ethamount 100 --to $user --wait
```
Comment (sveitser) [inline smoke-test-altlayer.bash]: I think by default the L2 url is set to the sequencer, not the full node

https://github.com/EspressoSystems/nitro-testnode/blob/b33fc8d47e964e49150e9fea6ecdffe733090b56/scripts/index.ts#L29

Comment (Sneh1999) [reply smoke-test-altlayer.bash]: Nice Catch - thank you! (https://github.com/EspressoSystems/nitro-testnode/pull/59/commits/1e6e2b43984682b894a6bcf3153db79c632f0124)

**Review by sveitser — APPROVED** [review]

**Pull request merged**